### PR TITLE
fix #80042 flipkart.com (mobile)

### LINKED ISSUE #80042: www.flipkart.com
### Issue URL (Ads)
 [https://www.flipkart.com/all/pr?sid=all&offer=nb%3Amp%3A034e4a2303&p%5B0%5D=facets.availability%255B%255D%3DExclude%2BOut%2Bof%2BStock&offerCategory=BuyMoreSaveMore&preOrderEnabled=false](https://adguardteam.github.io/AnonymousRedirect/redirect.html?url=https%3A%2F%2Fwww.flipkart.com%2Fall%2Fpr%3Fsid%3Dall%26offer%3Dnb%253Amp%253A034e4a2303%26p%255B0%255D%3Dfacets.availability%25255B%25255D%253DExclude%252BOut%252Bof%252BStock%26offerCategory%3DBuyMoreSaveMore%26preOrderEnabled%3Dfalse)
### Comment
Username: @chirag127
> ad leftovers

### Screenshots
<details>
  <summary>Screenshot 1</summary>

  Screenshot 1
</details>
<details>
  <summary>Screenshot 2</summary>

  Screenshot 2
</details>
<details>
  <summary>Screenshot 3</summary>

  Screenshot 3
</details>

### System configuration

Information | value
--- | ---
AdGuard product: | AdGuard Browser extension v3.5.34
Browser: | Chrome
Stealth mode options: | Hide your search queries, <br>Remove X-Client-Data header from HTTP requests, <br>Strip URLs from tracking parameters, <br>Self-destructing third-party cookies (2880), <br>Hide your Referrer from third-parties
Filters: | <b>Ad Blocking:</b><br/> AdGuard Base<br/><br/><b>Privacy:</b><br/> AdGuard Tracking Protection<br/><br/><b>Social Widgets:</b><br/> AdGuard Social Media<br/><br/><b>Annoyances:</b><br/> AdGuard Annoyances

### PR DESCRIPTION
**Pull request by wout-codes**
#80042

### CODE REVIEW

File: AnnoyancesFilter/sections/self-promo.txt [hunk 1]
```diff
@@ -340,9 +340,10 @@ nero.com###fancybox-wrap
 nero.com###fancybox-overlay
 ! https://github.com/AdguardTeam/AdguardFilters/issues/44364
 freiepresse.de###aboBanner
-! https://github.com/AdguardTeam/AdguardFilters/issues/44090
+! https://github.com/AdguardTeam/AdguardFilters/issues/80042
 flipkart.com##div[data-id][style] > div[data-tkid^="en_"]
 flipkart.com#?#div[style] > div[class*="col-"] > div[class] > div[data-id][style]:has(> div[data-tkid^="en_"])
+flipkart.com#?#div[class^="css-"] > div:has(> div[dir]:contains(/^AD$/))
```
Comment (stanislav-atr) [inline AnnoyancesFilter/sections/self-promo.txt:346]: As i remember correctly, there was a case of such elements and we decided not to block them, as they are paid by users and don't show up somewhere else again after blocking. I might be wrong, @Sergey-Lyapin, check this please.
<details><summary>Screenshot:</summary>

image</details><br/>

Comment (wout-codes) [reply AnnoyancesFilter/sections/self-promo.txt:346]: What do you mean by "as they are paid by users"?

Comment (Sergey-Lyapin) [reply AnnoyancesFilter/sections/self-promo.txt:346]: I see these elements are blocked as self-promo by Blaz

```
! https://github.com/AdguardTeam/AdguardFilters/issues/44090
flipkart.com##div[data-id][style] > div[data-tkid^="en_"]
flipkart.com#?#div[style] > div[class*="col-"] > div[class] > div[data-id][style]:has(> div[data-tkid^="en_"])
```

Comment (wout-codes) [reply AnnoyancesFilter/sections/self-promo.txt:346]: Yes, the rules I have crafted are for mobile.

Comment (Sergey-Lyapin) [reply AnnoyancesFilter/sections/self-promo.txt:346]: what about something like this?
```
flipkart.com#?##_parentCtr_ > div[style^="transform: translate"][style*="width: 360px;"]:has(div[dir="auto"]:contains(/^AD$/))
```

File: AnnoyancesFilter/sections/mobile-app.txt [hunk 2]
```diff
@@ -42,6 +42,7 @@ pinterest.*##div[data-test-id="header"] div[style^="background-color: rgba(255,
 pinterest.*##div[data-test-id="footer"] div[style="max-width: 264px;"]
 hindustantimes.com##.openAppButton
 india.com##.open-app
+flipkart.com#?#div[class*=" "] > div[class]:has(> div > img[src*="flipkart_app_icon-"])
```
Comment (Sergey-Lyapin) [inline AnnoyancesFilter/sections/mobile-app.txt:45]: I don't see this element

File: SpywareFilter/sections/specific.txt [hunk 3]
```diff
@@ -43,6 +43,7 @@
 ||app.carnow.com/dealers/visitor_info?
 ||hondacarsrockhill.com/pixall/pix-ddc-fp.min.js
 ||servebom.com/event.js
+||flipkart.net/*/*/tracker/
```
Comment (Sergey-Lyapin) [inline SpywareFilter/sections/specific.txt:46]: `||pla-tk.flipkart.net^` to tracking_servers_firstparty

File: AnnoyancesFilter/sections/self-promo.txt (hunk 1 above)
Comment (wout-codes) [reply AnnoyancesFilter/sections/self-promo.txt:346]: Mine is already narrowed enough.

File: AnnoyancesFilter/sections/mobile-app.txt (hunk 2 above)
Comment (wout-codes) [reply AnnoyancesFilter/sections/mobile-app.txt:45]: Will you merge if I show you?

File: AnnoyancesFilter/sections/self-promo.txt (hunk 1 above)
Comment (wout-codes) [reply AnnoyancesFilter/sections/self-promo.txt:346]: And this rule doesn't work on my end.

File: AnnoyancesFilter/sections/mobile-app.txt (hunk 2 above)
Comment (Sergey-Lyapin) [reply AnnoyancesFilter/sections/mobile-app.txt:45]: @wout-codes most likely

**Review by Alex-302 — APPROVED** [review]

**Pull request merged**